src/orion/storage/legacy.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.storage.legacy` -- Legacy storage
+=============================================================================
+
+.. module:: legacy
+   :platform: Unix
+   :synopsis: Old Storage implementation
+
+"""
+
+from orion.core.io.convert import JSONConverter
+from orion.core.io.database import Database
+from orion.core.worker.trial import Trial
+from orion.storage.base import BaseStorageProtocol
+
+
+class Legacy(BaseStorageProtocol):
+    """Legacy protocol, store all experiments and trials inside the Database()
+
+    Parameters
+    ----------
+    uri: str
+        database uri specifying how to connect to the database
+        the uri follows the following format
+        `mongodb://[username:password@]host1[:port1][,...hostN[:portN]][/[database][?options]]`
+
+    """
+
+    def __init__(self, uri=None):
+        self._db = Database()
+        self._setup_db()
+
+    def _setup_db(self):
+        """Database index setup"""
+        self._db.ensure_index('experiments',
+                              [('name', Database.ASCENDING),
+                               ('metadata.user', Database.ASCENDING)],
+                              unique=True)
+        self._db.ensure_index('experiments', 'metadata.datetime')
+
+        self._db.ensure_index('trials', 'experiment')
+        self._db.ensure_index('trials', 'status')
+        self._db.ensure_index('trials', 'results')
+        self._db.ensure_index('trials', 'start_time')
+        self._db.ensure_index('trials', [('end_time', Database.DESCENDING)])
+
+    def create_experiment(self, config):
+        """See :func:`~orion.storage.BaseStorageProtocol.create_experiment`"""
+        return self._db.write('experiments', config)
+
+    def update_experiment(self, experiment, where=None, **kwargs):
+        """See :func:`~orion.storage.BaseStorageProtocol.update_experiment`"""
+
+        if not kwargs:
+            return True
```

Comment:
return False?

Reply:
nothing to update is it really failing ?

Reply:
Well, if the question is *was the storage updated*, then the answer is no.